pkg/gitprovider/aws-codecommit.go

```diff
@@ -0,0 +1,298 @@
+		Username: *user.User.UserName,
+		Name:     *user.User.Arn,
+		Email:    "",
+	}, nil
+
+}
+
+func (g *AwsCodeCommitGitProvider) GetRepoPRs(repositoryId string, namespaceId string) ([]*GitPullRequest, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get client: %s", err.Error())
+	}
+	pullrequests, err := client.ListPullRequests(context.TODO(), &codecommit.ListPullRequestsInput{
+		RepositoryName:    &repositoryId,
+		PullRequestStatus: types.PullRequestStatusEnumOpen,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pull requests: %s", err.Error())
+	}
+	var pullRequests []*GitPullRequest
+	for _, pullrequestid := range pullrequests.PullRequestIds {
+		pr, err := client.GetPullRequest(context.TODO(), &codecommit.GetPullRequestInput{
+			PullRequestId: &pullrequestid,
+		})
+		if err != nil {
+			log.Warnf("failed to get pull request: %s error: %f", pullrequestid, err)
+		}
+		prbranch := strings.TrimPrefix(*pr.PullRequest.PullRequestTargets[0].SourceReference, "refs/heads/")
+		pullRequest := &GitPullRequest{
+			Name:            *pr.PullRequest.Title,
+			Branch:          *pr.PullRequest.PullRequestTargets[0].SourceReference,
+			Sha:             *pr.PullRequest.PullRequestTargets[0].SourceReference,
+			SourceRepoId:    prbranch,
+			SourceRepoName:  prbranch,
+			SourceRepoUrl:   getCodeCommitCloneUrl(g.region, repositoryId),
+			SourceRepoOwner: strings.Split(*pr.PullRequest.AuthorArn, ":")[5],
+		}
+		pullRequests = append(pullRequests, pullRequest)
+	}
+	return pullRequests, nil
+}
+
+func (g *AwsCodeCommitGitProvider) GetLastCommitSha(staticContext *StaticGitContext) (string, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return "", fmt.Errorf("failed to get client: %s", err.Error())
+	}
+	branch, err := client.GetBranch(context.TODO(), &codecommit.GetBranchInput{
+		RepositoryName: &staticContext.Name,
+		BranchName:     staticContext.Branch,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch branch details: %w", err)
+	}
+	return *branch.Branch.CommitId, nil
+}
+
+func (g *AwsCodeCommitGitProvider) getPrContext(staticContext *StaticGitContext) (*StaticGitContext, error) {
+	if staticContext.PrNumber == nil {
+		return staticContext, nil
+	}
+	client, err := g.getApiClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get client: %s", err.Error())
+	}
+	prnumber := strconv.FormatUint(uint64(*staticContext.PrNumber), 10)
+	pr, err := client.GetPullRequest(context.TODO(), &codecommit.GetPullRequestInput{
+		PullRequestId: aws.String(prnumber),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pull request: %s", err.Error())
+	}
+	repo := *staticContext
+	prbranchname := strings.TrimPrefix(*pr.PullRequest.PullRequestTargets[0].SourceReference, "refs/heads/")
+	repo.Branch = &prbranchname
+	return &repo, nil
+}
+
+func (g *AwsCodeCommitGitProvider) parseStaticGitContext(repoUrl string) (*StaticGitContext, error) {
```

Comment:
Something is wrong with parsing context from a commit url.

`dtn create https://us-east-1.console.aws.amazon.com/codesuite/codecommit/repositories/test/commit/d6921fe5f1fb73ccaca4ba902528ee8e3920e507\?region\=us-east-1 -t local` when I ran this, I got this context:

```
"repository": {
        "branch": "main",
        "id": "test",
        "name": "test",
        "owner": "test",
        "sha": "0bf7cc6a744332c4d99b7397e58044e1aa84ec5d",
        "source": "us-east-1.console.aws.amazon.com",
        "url": "https://git-codecommit.us-east-1.amazonaws.com/v1/repos/test"
      },
```

Notice that the commit sha is wrong. The expected behavior is that sha and branch are set to `d6921fe5f1fb73ccaca4ba902528ee8e3920e507`

Reply:
`daytona info -o json WORKSPACE_NAME`

Reply:
```
daytona ➜ /workspaces/daytona (03795951) $ dtn create https://us-east-1.console.aws.amazon.com/codesuite/codecommit/repositories/test/commit/d6921fe5f1fb73ccaca4ba902528ee8e3920e507\?region\=us-east-1    
FATA[0000] failed to get repository: failed to fetch branch details: operation error CodeCommit: GetBranch, https response error StatusCode: 400, RequestID: cab4d36c-880f-4e54-93f5-0d7df4ab9212, BranchDoesNotExistException:  
exit status 1
```

This is the error that I now get when trying to create from a commit URL